From 588c94e6ad4c8ce5278fd063e20a39c1cb8f67db Mon Sep 17 00:00:00 2001
From: Ulises Mendez Martinez <ulises.mendez.martinez@intel.com>
Date: Fri, 8 Apr 2016 15:02:56 -0500
Subject: [PATCH 3/5] iio: magn: ak0911: expose available frequencies

This patch will add the attribute missing to
expose the frequencies available for sensor.

Change-Id: Ib1856249f5cdb5d0d330b03f8b280a7ca1bdd3dc
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-28469
Signed-off-by: Ulises Mendez Martinez <ulises.mendez.martinez@intel.com>
Reviewed-on: https://android.intel.com:443/492622
---
 drivers/iio/magnetometer/ak09911.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/iio/magnetometer/ak09911.c b/drivers/iio/magnetometer/ak09911.c
index 84e4916..783389e 100644
--- a/drivers/iio/magnetometer/ak09911.c
+++ b/drivers/iio/magnetometer/ak09911.c
@@ -328,6 +328,17 @@ static int ak09911_read_raw(struct iio_dev *indio_dev,
 	return -EINVAL;
 }
 
+static IIO_CONST_ATTR_SAMP_FREQ_AVAIL("10 20 50 100");
+
+static struct attribute *ak09911_attributes[] = {
+	&iio_const_attr_sampling_frequency_available.dev_attr.attr,
+	NULL,
+};
+
+static const struct attribute_group ak09911_attrs_group = {
+	.attrs = ak09911_attributes,
+};
+
 enum ak09911_axis {
 	AXIS_X = 0,
 	AXIS_Y,
@@ -419,6 +430,7 @@ static int ak09911_write_raw(struct iio_dev *indio_dev,
 }
 
 static const struct iio_info ak09911_info = {
+	.attrs		= &ak09911_attrs_group,
 	.read_raw	= ak09911_read_raw,
 	.write_raw	= ak09911_write_raw,
 	.driver_module	= THIS_MODULE,
-- 
1.9.1

